src/topology/algebra/ordered.lean

```diff
@@ -2198,22 +2198,185 @@ lemma tendsto_of_monotone {ι α : Type*} [preorder ι] [topological_space α]
 if H : bdd_above (range f) then or.inr ⟨_, tendsto_at_top_csupr h_mono H⟩
 else or.inl $ tendsto_at_top_at_top_of_monotone' h_mono H
 
-@[to_additive] lemma tendsto_inv_nhds_within_Ioi {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Ioi [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within a (Ioi a)) (nhds_within (a⁻¹) (Iio (a⁻¹))) :=
 (continuous_inv.tendsto a).inf $ by simp [tendsto_principal_principal]
 
-@[to_additive] lemma tendsto_inv_nhds_within_Iio {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Iio [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within a (Iio a)) (nhds_within (a⁻¹) (Ioi (a⁻¹))) :=
 (continuous_inv.tendsto a).inf $ by simp [tendsto_principal_principal]
 
-@[to_additive] lemma tendsto_inv_nhds_within_Ioi_inv {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Ioi_inv [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within (a⁻¹) (Ioi (a⁻¹))) (nhds_within a (Iio a)) :=
 by simpa only [inv_inv] using @tendsto_inv_nhds_within_Ioi _ _ _ _ (a⁻¹)
 
-@[to_additive] lemma tendsto_inv_nhds_within_Iio_inv {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Iio_inv [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within (a⁻¹) (Iio (a⁻¹))) (nhds_within a (Ioi a)) :=
 by simpa only [inv_inv] using @tendsto_inv_nhds_within_Iio _ _ _ _ (a⁻¹)
+
+lemma continuous_at_iff_continuous_left_right [topological_space α] [linear_order α]
+  [topological_space β] {a : α} {f : α → β} :
+  continuous_at f a ↔ continuous_within_at f (Iic a) a ∧ continuous_within_at f (Ici a) a :=
+begin
+  split,
+  exact λ h, ⟨ h.continuous_within_at, h.continuous_within_at ⟩,
+  rintros ⟨ h₁, h₂ ⟩,
+  apply (continuous_within_at_univ _ _).mp,
+  convert ← h₁.union h₂,
+  simp only [h₁.union h₂, eq_univ_iff_forall, Ici, Iic, le_total, mem_union_eq, mem_set_of_eq, forall_true_iff]
+end
+
+lemma continuous_on_Ico_extend_continuous_on_Ioo
+  [topological_space α] [linear_order α] [order_topology α] [topological_space β] {a b : α} {l : β}
+  {f : α → β} (hcf : continuous_on f (Ioo a b)) (hfa : tendsto f (nhds_within a (Ioi a)) (nhds l)) :
+  continuous_on (λ x, if x ∈ Ioo a b then f x else l) (Ico a b) :=
+begin
+  by_cases hab : a < b,
+  { have : ∀ x ∈ frontier (Ioo a b) ∩ Ico a b, x = a,
+    { rintros x ⟨ ⟨ _, hx₁ ⟩, hx₂ ⟩,
+      rw interior_Ioo at hx₁,
+      simp only [mem_Ioo, not_and, not_lt] at hx₁,
+      by_cases h : a = x,
+      { simp [h] },
+      { exfalso,
+        exact not_le_of_lt hx₂.2 (hx₁ $ lt_of_le_of_ne hx₂.1 h) } },
+    apply continuous_on_if,
+    repeat
+    { intros a' ha',
+      rw (show (λ (a_1 : α), a_1 ∈ Ioo a b) = Ioo a b, from rfl) at *,
```

Comment:
This is fishy. If you need such incantations then something is wrong somewhere.

Reply:
Indeed, I was working on making it cleaner. It seems like mixing `α -> Prop` and `set α` leads to notation nightmare

Reply:
Yes, this is a recurring theme. That's why we sometimes have two versions of the same lemma, one using `α -> Prop`
 and one using `set α`.